File: codegen/results/v002/qwen3.5_4b-nvfp4_think/20260412_083548/python/008_prime_numbers/solution.py
import sys

def is_prime(n):
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    
    for i in range(3, int(n**0.5) + 1, 2):
        if n % i == 0:
            return False
    return True

def main():
    args = sys.argv[1:]
    if len(args) != 1:
        print("Usage: python script.py <N>")
        return
    
    try:
        n = int(args[0])
    except ValueError:
        print("Error: Please provide a valid integer")
        return
    
    for i in range(2, n + 1):
        if is_prime(i):
            print(i)

if __name__ == "__main__":
    main()

File: codegen/results/v002/qwen3.5_4b-nvfp4_think/20260412_083548/go/008_prime_numbers/solution.go
package main

import (
	"fmt"
	"os"
)

func main() {
	if len(os.Args) != 2 {
		fmt.Fprintln(os.Stderr, "Usage: go run . <number>")
		os.Exit(1)
	}

	n, err := strconv.Atoi(os.Args[1])
	if err != nil || n < 0 {
		fmt.Fprintln(os.Stderr, "Error: Please provide a non-negative integer")
		os.Exit(1)
	}

	// Handle numbers below 2 where no primes exist
	if n < 2 {
		return
	}

	// Use Sieve of Eratosthenes to find primes up to N
	initialized := make([]bool, n+1)
	for i := 2; i <= n; i++ {
		initialized[i] = true
	}

	for i := 2; i*i <= n; i++ {
		if initialized[i] {
			for j := i * i; j <= n; j += i {
				initialized[j] = false
			}
		}
	}

	for i := 2; i <= n; i++ {
		if initialized[i] {
			fmt.Println(i)
		}
	}
}